src/components/common/Button/index.tsx

```diff
@@ -0,0 +1,95 @@
+import styled, { ThemedStyledProps } from 'styled-components'
+import { ColourTheme, ThemeType } from 'styles/theme'
+
+const { MAIN, SECONDARY } = ThemeType
+
+export const ButtonBase = styled.button.attrs(({ $alt, $border, theme }: { // $alt button theme toggle
+  $alt?: boolean; $border?: boolean; theme: ThemedStyledProps<ColourTheme, ColourTheme> }) => ({
+  // Handle button theming
+  button: {
+    bg1: !$alt ? theme.button[MAIN].bg1 : theme.button[SECONDARY].bg1,
+    bg1Hover: !$alt ? theme.button[MAIN].bg1Hover : theme.button[SECONDARY].bg1Hover,
+    text1: !$alt ? theme.button[MAIN].text1 : theme.button[SECONDARY].text1,
+    text1Hover: !$alt ? theme.button[MAIN].text1Hover : theme.button[SECONDARY].text1Hover,
+  },
+  // Show/Hide border toggle
+  $border,
+}))`
+  border: 0.1rem solid;
+  border-color: ${({ $border, button }): string => ($border ? button.text1 : 'transparent')};
+  background-color: ${({ button }): string => button.bg1};
+  color: ${({ button }): string => button.text1};
+
+  &:hover {
+    border-color: ${({ button }): string => button.bg1Hover};
+    background-color: ${({ button }): string => button.bg1Hover};
+    color: ${({ button }): string => button.text1Hover};
+  }
+
+  border-radius: 2rem;
+  cursor: pointer;
+  font-weight: bold;
+  outline: 0;
+  padding: 0.5rem 1rem;
+
+  transition: all 0.2s ease-in-out;
+  transition-property: color, background-color, border-color, opacity;
+
+  &.cancel {
+    background: transparent;
+    color: var(--color-text-active);
```

Comment:
I'm really not a fan of how we mix styled-comp theme and css vars theme, but whatever

Reply:
Me neither, will rework for prop based logic 🤙